### PR TITLE
Add Mongo scaler documentation

### PR DESCRIPTION
**Pull request by NUCsimple**
This PR adds documentation for Mongo scaler.
Relates to kedacore/keda#1465.

### CODE REVIEW

**Comment by tomkerkhove** [comment]
@NUCsimple Can you fix DCO please?

File: content/docs/2.1/scalers/mongo.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# mongo with ScaledJob
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:1]: Can you add the typical scaler header please?

https://github.com/kedacore/keda-docs/blob/master/content/docs/2.1/scalers/azure-service-bus.md#L1-L8

Feel free to drop the reference to `ScaledJob`

File: content/docs/2.1/scalers/mongo.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+## Trigger Specification
+This specification describes the mongo trigger that scales based on result of Mongo query.
+
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:5]: Can you add a summary similar to https://github.com/kedacore/keda-docs/blob/master/content/docs/2.1/scalers/azure-service-bus.md#L14 here please?

File: content/docs/2.1/scalers/mongo.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# mongo with ScaledJob
+
+## Trigger Specification
+This specification describes the mongo trigger that scales based on result of Mongo query.
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:4]:
```suggestion
This specification describes the `mongo` trigger that scales based on result of Mongo query.
```

File: content/docs/2.1/scalers/mongo.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+The trigger always requires the following information:
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:6]: Can you align with the approach from https://github.com/kedacore/keda-docs/blob/master/content/docs/2.1/scalers/azure-service-bus.md#L31-L38 please?

File: content/docs/2.1/scalers/mongo.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+Or provide more detailed information:
+- host - The host of the Mongo server
+- port - The port of the Mongo server
+- username - Username to authenticate with to Mongo database
+- passwordFromEnv - Password for the given user
+## Authentication Parameters 
+You can authenticate by using connection string or password authentication.
+
+Connection String Authentication:
+- connectionString  -Connection string for Mongo database
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:24]:
```suggestion
- `connectionString` - Connection string for Mongo database
```

File: content/docs/2.1/scalers/mongo.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Password Authentication:
+- password - Password for configured user to login to Mongo database variables.
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:27]:
```suggestion
- `password` - Password for the configured user to login to Mongo database variables.
```

File: content/docs/2.1/scalers/mongo.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+- port - The port of the Mongo server
+- username - Username to authenticate with to Mongo database
+- passwordFromEnv - Password for the given user
+## Authentication Parameters 
+You can authenticate by using connection string or password authentication.
+
+Connection String Authentication:
+- connectionString  -Connection string for Mongo database
+
+Password Authentication:
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:26]:
```suggestion
**Password Authentication:**

```

File: content/docs/2.1/scalers/mongo.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+Or provide more detailed information:
+- host - The host of the Mongo server
+- port - The port of the Mongo server
+- username - Username to authenticate with to Mongo database
+- passwordFromEnv - Password for the given user
+## Authentication Parameters 
+You can authenticate by using connection string or password authentication.
+
+Connection String Authentication:
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:23]:
```suggestion
**Connection String Authentication:**

```

File: content/docs/2.1/scalers/mongo.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+To provide information about how to connect to Mongo you can provide
+- connectionStringFromEnv -Mongo connection string that should point to environment variable with valid value
+
+Or provide more detailed information:
+- host - The host of the Mongo server
+- port - The port of the Mongo server
+- username - Username to authenticate with to Mongo database
+- passwordFromEnv - Password for the given user
+## Authentication Parameters
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongo.md:20]:
```suggestion

## Authentication Parameters

```

**Review by tomkerkhove — COMMENTED** [review]
It's a good start but needs a bit of alignment with our other scalers, thanks!

Also, can you change all `mongo` references to `mongoDb` please?

File: content/docs/2.1/scalers/mongo.md (hunk 9 above)
Comment (NUCsimple) [reply content/docs/2.1/scalers/mongo.md]: I think it should be ok now. @tomkerkhove ,pls review it again.

File: content/docs/2.1/scalers/mongoDB.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+layout = "scaler"
+maintainer = "Community"
+description = "Scale applications based on mongoDB."
+availability = "v2.1+"
+go_file = "mongoDB_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the mongoDB trigger that scales based on result of mongoDB query.
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongoDB.md:12]: A big nit but can you replace all `mongoDB` with `MongoDB` please? (except for trigger type references of course.

File: content/docs/2.1/scalers/mongoDB.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+      query: '{"region":"eu-1","state":"running","plan":"planA"}'
+      # Required: according to the number of query result, to scale job
+      queryValue: "1"
+```
+
+**Parameter list:**
+
+- `dbName`  - Name of the database
+- `collection` - Name of the collection
+- `query` - a mongoDB query that should return single numeric value
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongoDB.md:32]:
```suggestion
- `query` - A MongoDB query that should return single numeric value
```

File: content/docs/2.1/scalers/mongoDB.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+- `queryValue` - a threshold that is used as targetAverageValue in HPA.
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongoDB.md:33]:
```suggestion
- `queryValue` - A threshold that will define when scaling should occur.
```

File: content/docs/2.1/scalers/mongoDB.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+To provide information about how to connect to mongoDB you can provide
+
+- `connectionStringFromEnv` - mongoDB connection string that should point to environment variable with valid value
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongoDB.md:37]:
```suggestion
- `connectionStringFromEnv` - A MongoDB connection string that should point to environment variable with valid value
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
LGTM, just need a small change with respect to the product name - Sorry!

File: content/docs/2.1/scalers/mongoDB.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+layout = "scaler"
+maintainer = "Community"
+description = "Scale applications based on MongoDB."
+availability = "v2.1+"
+go_file = "MongoDB_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the MongoDB trigger that scales based on result of MongoDB query.
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/mongoDB.md:12]: We typically use the trigger type here:
```suggestion
This specification describes the `mongoDB` trigger that scales based on result of MongoDB query.
```

**Review by tomkerkhove — APPROVED** [review]
LGTM, thanks! Just waiting for the feature to be merged now.

File: content/docs/2.1/scalers/mongoDB.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+layout = "scaler"
+maintainer = "Community"
+description = "Scale applications based on MongoDB."
+availability = "v2.1+"
+go_file = "MongoDB_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `mongoDB` trigger that scales based on result of MongoDB query.
```
Comment (zroubalik) [inline content/docs/2.1/scalers/mongoDB.md:12]:
```suggestion
This specification describes the `mongodb` trigger that scales based on result of MongoDB query.
```

**Review by zroubalik — CHANGES_REQUESTED** [review]
There are minor details to be fixed

**Comment by tomkerkhove** [comment]
Any update on this @NUCsimple?

**Comment by zroubalik** [comment]
@NUCsimple could you please fix this, so we can proceed with a release? Thanks!

**Comment by NUCsimple** [comment]
@zroubalik I'll fix it today,sorry.

File: content/docs/1.5/scalers/mongodb.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+      metricName: "global_metric"
+```
+
+**Parameter list:**
+
+- `dbName`  - Name of the database
+- `collection` - Name of the collection
+- `query` - A MongoDB query that should return single numeric value
+- `queryValue` - A threshold that will define when scaling should occur
+- `metricName` - An optional name to assign to the metric.
```
Comment (zroubalik) [inline content/docs/1.5/scalers/mongodb.md:36]: Information about `metricName` is missing, please take a look on example info that's provided by other scalers:
https://keda.sh/docs/2.1/scalers/influxdb/ and https://keda.sh/docs/2.1/scalers/postgresql/

**Review by zroubalik — COMMENTED** [review]
Looking good, there's minor nit in the `metricName` documentation

@NUCsimple could you please add this?

**Comment by zroubalik** [comment]
@NUCsimple  I've just noticed that you placed the scaler docs in `content/docs/1.5/scalers/mongodb.md` could you please move it to `content/docs/2.1/scalers/mongodb.md`, as this scaler will be present from KEDA 2.1 onwards

**Comment by NUCsimple** [comment]
@zroubalik  ,I'm so sorry. Maybe the mouse moved the file by mistake just now

**Review by zroubalik — APPROVED** [review]

**Pull request merged**